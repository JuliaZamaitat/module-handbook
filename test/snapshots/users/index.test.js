const { app, User, request } = require('../../commonJest')
const userData = require('../../../mongo/seed/users')
const ObjectID = require('mongodb').ObjectID
function to12 (s) {
  if (s.length > 12) {
    return s.substr(0, 12)
  }
  return s + '_'.repeat(12 - s.length)
}
describe('User Index', function () {
  beforeEach(function (done) {
    const dataFixedIDs = userData.map(u => {
      u._id = new ObjectID(to12(u.email))
      return u
    })

    User.create(dataFixedIDs)
    // to see tests fail, use this:
    // return Course.create([{ code: 'dummy' }])
      .then(createdCourses => {
        // console.log(createdCourses.length + ' courses created')
        done()
      })
      .catch(error => {
        console.log('error caught: ' + error.message)
        done(error.message)
      })
  })
  it('renders user index correctly', (done) => {
    request(app)
      .get('/users')
      .then((response) => {
        expect(response.text).toMatchSnapshot()
        done()
      })
  })
})
