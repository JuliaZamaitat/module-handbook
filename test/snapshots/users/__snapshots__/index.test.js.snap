// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`User Index renders user index correctly 1`] = `
"<!DOCTYPE html>
<html>

<head>
	<meta charset=\\"utf-8\\">
	<meta name=\\"viewport\\" content=\\"width=device-width, initial-scale=0.8, maximum-scale=1\\">
	<title>Module Handbook</title>
	<link rel=\\"shortcut icon\\" type=\\"image/png\\" href=\\"/images/icon.png\\" />
	<link rel=\\"stylesheet\\" href=\\"/css/bootstrap.css\\">
	<link rel=\\"stylesheet\\" href=\\"/css/module_handbook.css\\">
	<script src=\\"/jquery.min.js\\"></script>
	<!--<script src=\\"/popper.js\\"></script>-->
	<script src=\\"/js/bootstrap.min.js\\"></script>

</head>

<body>
	<div id=\\"nav\\">
		<div class=\\"col-sm nav-align\\">
			<h1 id=\\"title\\">Module Handbook</h1>
		</div>
		<nav class=\\"navbar navbar-expand-lg navbar-light bg-light\\">
    <a class=\\"navbar-brand\\" href=\\"/\\">Home</a>
    <button class=\\"navbar-toggler\\" type=\\"button\\" data-toggle=\\"collapse\\" data-target=\\"#navbarSupportedContent\\"
            aria-controls=\\"navbarSupportedContent\\" aria-expanded=\\"false\\" aria-label=\\"Toggle navigation\\">
        <span class=\\"navbar-toggler-icon\\"></span>
    </button>

    <div class=\\"collapse navbar-collapse\\" id=\\"navbarSupportedContent\\">
        <ul class=\\"navbar-nav mr-auto\\">
            
                <li class=\\"nav-item \\">
    <a class=\\"nav-link\\" href=\\"/modules/list\\">Module List
        
    </a>
</li>

            
                <li class=\\"nav-item \\">
    <a class=\\"nav-link\\" href=\\"/modules/tabular\\">Module Table
        
    </a>
</li>

            
                <li class=\\"nav-item \\">
    <a class=\\"nav-link\\" href=\\"/courses\\">Courses
        
    </a>
</li>

            
                <li class=\\"nav-item active\\">
    <a class=\\"nav-link\\" href=\\"/users\\">Users
        <span class=\\"sr-only\\">(current)</span>
    </a>
</li>

            
                <li class=\\"nav-item \\">
    <a class=\\"nav-link\\" href=\\"/about\\">About
        
    </a>
</li>

            

        </ul>
        <form class=\\"form-inline my-2 my-lg-0\\" action = \\"/about\\" method = \\"POST\\">
            <input class=\\"form-control mr-sm-2\\" type=\\"text\\" placeholder=\\"Search\\" aria-label=\\"Search\\" name = \\"query\\">
            <button class=\\"btn btn-outline-success my-2 my-sm-0\\" type=\\"submit\\">Search</button>
        </form>
    </div>
</nav>
/users

	</div>

	<h2 class=\\"center\\">Users Table</h2>
<div class=\\"center\\">
  <a class=\\"button\\" href=\\"/users/new\\">Create User</a>
</div>
<table class=\\"table\\">
  <thead>
    <tr>
      <th>Name</th>
      <th>Email</th>
      <th>Zip Code</th>
      <th>Edit</th>
      <th>Delete</th>
    </tr>
  </thead>
  <tbody>
    
    <tr>
      <td>
        <a href=\\"/users/646f7269616e5f626f726773\\">
          Dorian Borgschulze
        </a>
      </td>
      <td>
        dorian_borgschulze@gmail.com
      </td>
      <td>
        57138
      </td>
      <td>
        <a href=\\"/users/646f7269616e5f626f726773/edit\\">
          Edit
        </a>
      </td>
      <td>
        <a href=\\"/users/646f7269616e5f626f726773/delete?_method=DELETE\\" onclick=\\"return confirm('Are you sure you want to delete this record?')\\" a>Delete</a>
      </td>
    </tr>
    
    <tr>
      <td>
        <a href=\\"/users/73756d6d6572383840796168\\">
          Cecile Kreutz
        </a>
      </td>
      <td>
        summer88@yahoo.com
      </td>
      <td>
        35525
      </td>
      <td>
        <a href=\\"/users/73756d6d6572383840796168/edit\\">
          Edit
        </a>
      </td>
      <td>
        <a href=\\"/users/73756d6d6572383840796168/delete?_method=DELETE\\" onclick=\\"return confirm('Are you sure you want to delete this record?')\\" a>Delete</a>
      </td>
    </tr>
    
    <tr>
      <td>
        <a href=\\"/users/696c617964615f6869727363\\">
          Ilayda Kulma
        </a>
      </td>
      <td>
        ilayda_hirsch81@gmail.com
      </td>
      <td>
        36953
      </td>
      <td>
        <a href=\\"/users/696c617964615f6869727363/edit\\">
          Edit
        </a>
      </td>
      <td>
        <a href=\\"/users/696c617964615f6869727363/delete?_method=DELETE\\" onclick=\\"return confirm('Are you sure you want to delete this record?')\\" a>Delete</a>
      </td>
    </tr>
    
  </tbody>
</table>


</body>

</html>
"
`;
